In [1]:
import pandas as pd # pandas use for loading  and manipulating the data

**Load the Dataset**

In [5]:
df=pd.read_csv("C:\\Users\\Abc\\Downloads\\student_dataset.csv")
df

StudentID    Name  Gender           Course  Age  Marks Grade       City
0         2001    Isha  Female     Data Science   27     82     F  Bangalore
1         2002   Aarav  Female     Data Science   27     42     A  Hyderabad
2         2003  Vivaan    Male  Web Development   24     98     B    Kolkata
3         2004   Sanya  Female  Web Development   23     52     A  Hyderabad
4         2005   Anaya  Female  Web Development   28     51     A  Hyderabad
..         ...     ...     ...              ...  ...    ...   ...        ...
145       2146   Anaya    Male  Cloud Computing   19     44     B  Bangalore
146       2147   Meera  Female          AI & ML   18     57     D  Hyderabad
147       2148   Kabir  Female  Cloud Computing   21     92     F  Bangalore
148       2149   Sanya  Female   Cyber Security   26     60     D      Delhi
149       2150    Diya  Female          AI & ML   24     44     F      Delhi

[150 rows x 8 columns]

In [8]:
df.head() # to get first 5 rows in data set

StudentID    Name  Gender           Course  Age  Marks Grade       City
0       2001    Isha  Female     Data Science   27     82     F  Bangalore
1       2002   Aarav  Female     Data Science   27     42     A  Hyderabad
2       2003  Vivaan    Male  Web Development   24     98     B    Kolkata
3       2004   Sanya  Female  Web Development   23     52     A  Hyderabad
4       2005   Anaya  Female  Web Development   28     51     A  Hyderabad

In [10]:
df.tail() # to get last 5 rows

StudentID   Name  Gender           Course  Age  Marks Grade       City
145       2146  Anaya    Male  Cloud Computing   19     44     B  Bangalore
146       2147  Meera  Female          AI & ML   18     57     D  Hyderabad
147       2148  Kabir  Female  Cloud Computing   21     92     F  Bangalore
148       2149  Sanya  Female   Cyber Security   26     60     D      Delhi
149       2150   Diya  Female          AI & ML   24     44     F      Delhi

In [12]:
df.describe(include='all') # to describe the data both categorical and numerical

StudentID   Name Gender        Course         Age       Marks Grade  \
count    150.000000    150    150           150  150.000000  150.000000   150   
unique          NaN     10      2             5         NaN         NaN     5   
top             NaN  Anaya   Male  Data Science         NaN         NaN     A   
freq            NaN     21     77            36         NaN         NaN    36   
mean    2075.500000    NaN    NaN           NaN   24.346667   67.033333   NaN   
std       43.445368    NaN    NaN           NaN    3.701962   17.598206   NaN   
min     2001.000000    NaN    NaN           NaN   18.000000   40.000000   NaN   
25%     2038.250000    NaN    NaN           NaN   21.250000   52.000000   NaN   
50%     2075.500000    NaN    NaN           NaN   24.000000   62.000000   NaN   
75%     2112.750000    NaN    NaN           NaN   27.000000   82.750000   NaN   
max     2150.000000    NaN    NaN           NaN   30.000000  100.000000   NaN   

             City  
count         150  
unique          5  
top     Bangalore  
freq           48  
mean          NaN  
std           NaN  
min           NaN  
25%           NaN  
50%           NaN  
75%           NaN  
max           NaN

In [14]:
df.describe() # to get statistical summary of only numerical column

StudentID         Age       Marks
count   150.000000  150.000000  150.000000
mean   2075.500000   24.346667   67.033333
std      43.445368    3.701962   17.598206
min    2001.000000   18.000000   40.000000
25%    2038.250000   21.250000   52.000000
50%    2075.500000   24.000000   62.000000
75%    2112.750000   27.000000   82.750000
max    2150.000000   30.000000  100.000000

**Student**
- There are 150 students record.
- The average studentID is 2075.5.
- maximum student are 2150.00
- minimum student are 2001.00
- The distribution of studentID deviation of about 43.44.

**Age**
- There is no null values in age column everyone has age.
- The average age is approximetly 24.35 years.
- The youngest student being 18 year old,and the oldest being 30.
- The most student are between 21.25 and 27 year  old, showing a moderately varied age range with standarad deviation of 3.7 year

**Marks**
- The average marks is 67.03
- The student who have lowest marks is 30.
- The highest marks is a perfect 100
- The standard deviation is 17.6 that shows most student scoring between 52 and 82.75.

In [17]:
df.shape # shows the number of rows and column

(150, 8)

**There are 150 rows and 8 column are present in data set.**

In [20]:
df.info() # shows the summary of dataframe

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 8 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   StudentID  150 non-null    int64 
 1   Name       150 non-null    object
 2   Gender     150 non-null    object
 3   Course     150 non-null    object
 4   Age        150 non-null    int64 
 5   Marks      150 non-null    int64 
 6   Grade      150 non-null    object
 7   City       150 non-null    object
dtypes: int64(3), object(5)
memory usage: 9.5+ KB


In [24]:
df.nunique() # that shows the unique values in data

StudentID    150
Name          10
Gender         2
Course         5
Age           13
Marks         56
Grade          5
City           5
dtype: int64

**There are 150 unique values in StudentID,and there are 10 unique name,2 genders are present,5 courses are available,13 ages,56 marks are preasent,5 unique grades are in dataset and 5 cities are available.**

In [27]:
df.isnull().sum() # isnull() to check for missing values and in order to get the total count of missing values  

StudentID    0
Name         0
Gender       0
Course       0
Age          0
Marks        0
Grade        0
City         0
dtype: int64

**There are no missing values are present in our dataset.**

In [30]:
df.rename(columns={'Course':'Designation'},inplace=True) # rename/change the column name

In [32]:
df.columns # display the column name

Index(['StudentID', 'Name', 'Gender', 'Designation', 'Age', 'Marks', 'Grade',
       'City'],
      dtype='object')

**Here we change the column name 'Course' to  'Designation'.**

In [49]:
df['Designation'].unique() # shows the unique values in Designation

array(['Data Science', 'Web Development', 'AI & ML', 'Cyber Security',
       'Cloud Computing'], dtype=object)

In [51]:
df.nunique()

StudentID      150
Name            10
Gender           2
Designation      5
Age             13
Marks           56
Grade            5
City             5
dtype: int64

In [53]:
df['Designation'].value_counts() # shows the value count in each category

Designation
Data Science       36
Web Development    35
AI & ML            34
Cyber Security     25
Cloud Computing    20
Name: count, dtype: int64

**There are 36 students in Data Science ,35 students in Web development,34 students in AI & ML ,25 students in Cyber security and 20 student whose designation is cloud computing.** 

In [56]:
df.groupby("Designation")["Marks"].mean() # to create group

Designation
AI & ML            67.911765
Cloud Computing    68.450000
Cyber Security     66.240000
Data Science       70.888889
Web Development    61.971429
Name: Marks, dtype: float64

- Students in Data Science have highest average marks i.e 67.911.
- Student in web development have lowest average marks i.e 61.97.


In [63]:
df.sort_values(by="Marks",ascending=False) 

StudentID   Name  Gender      Designation  Age  Marks Grade       City
17        2018  Aarav  Female     Data Science   27    100     C    Kolkata
36        2037   Diya  Female  Web Development   29    100     C  Bangalore
10        2011  Aarav  Female     Data Science   26     99     B  Hyderabad
12        2013   Isha  Female  Cloud Computing   20     99     D  Bangalore
69        2070   Diya  Female          AI & ML   23     99     F  Hyderabad
..         ...    ...     ...              ...  ...    ...   ...        ...
62        2063  Aarav    Male  Web Development   23     42     A     Mumbai
128       2129  Aarav  Female  Web Development   30     41     F     Mumbai
45        2046  Anaya    Male          AI & ML   30     41     A    Kolkata
130       2131  Meera    Male  Web Development   23     40     C  Bangalore
107       2108  Sanya    Male     Data Science   24     40     B     Mumbai

[150 rows x 8 columns]

**Here we sort the Marks in highest to lowest order.**

In [66]:
df.drop("StudentID",axis=1,inplace=True) # drop() is used to remove the column

In [68]:
df.drop(0,axis=0,inplace=True)  # dropping rows using index position,axis=0 for rows

In [70]:
df.index

RangeIndex(start=1, stop=150, step=1)

In [81]:
df[df['Marks'] > 90]

Name  Gender      Designation  Age  Marks Grade       City
2    Vivaan    Male  Web Development   24     98     B    Kolkata
5     Aarav    Male  Web Development   28     96     D     Mumbai
10    Aarav  Female     Data Science   26     99     B  Hyderabad
12     Isha  Female  Cloud Computing   20     99     D  Bangalore
17    Aarav  Female     Data Science   27    100     C    Kolkata
30    Rohan  Female     Data Science   24     92     D  Bangalore
36     Diya  Female  Web Development   29    100     C  Bangalore
46    Aryan    Male     Data Science   22     95     F  Hyderabad
47   Vivaan    Male  Web Development   25     91     C  Hyderabad
49     Isha    Male  Web Development   21     94     D  Bangalore
59    Anaya    Male   Cyber Security   18     95     F  Hyderabad
69     Diya  Female          AI & ML   23     99     F  Hyderabad
71    Aarav  Female  Cloud Computing   19     92     B    Kolkata
74    Rohan  Female  Cloud Computing   23     96     F  Bangalore
79    Rohan    Male  Web Development   24     94     D  Hyderabad
104    Isha  Female     Data Science   23     97     B  Bangalore
117   Sanya    Male     Data Science   18     91     D  Bangalore
119   Aryan  Female          AI & ML   25     96     A      Delhi
124   Kabir    Male   Cyber Security   26     97     D    Kolkata
131    Isha    Male     Data Science   30     99     B    Kolkata
147   Kabir  Female  Cloud Computing   21     92     F  Bangalore

In [78]:
df[df['Marks'] > 90].shape[0]# filtering the column base on conditon and shap[0]  that count the each student

21

**There are 21 students who have getting more than 90 marks.**

In [85]:
df.loc[4] # loc is use for label based indexing.returns rows with the index label 2

Name                     Anaya
Gender                  Female
Designation    Web Development
Age                         28
Marks                       51
Grade                        A
City                 Hyderabad
Name: 4, dtype: object

**This is shows the all values in 4 th rows. like Student name,gender,Designation,Age,marks,grade and city.**

In [89]:
df.loc[:,["Name","Marks"]] # this is shows only Selected column  values

Name  Marks
1     Aarav     42
2    Vivaan     98
3     Sanya     52
4     Anaya     51
5     Aarav     96
..      ...    ...
145   Anaya     44
146   Meera     57
147   Kabir     92
148   Sanya     60
149    Diya     44

[149 rows x 2 columns]

In [91]:
df.iloc[3] # returns the row for the specified index position . 

Name                     Anaya
Gender                  Female
Designation    Web Development
Age                         28
Marks                       51
Grade                        A
City                 Hyderabad
Name: 4, dtype: object

In [93]:
df.iloc[:, 0:10] # accessing the records for all the  columns with the index 0-10

Name  Gender      Designation  Age  Marks Grade       City
1     Aarav  Female     Data Science   27     42     A  Hyderabad
2    Vivaan    Male  Web Development   24     98     B    Kolkata
3     Sanya  Female  Web Development   23     52     A  Hyderabad
4     Anaya  Female  Web Development   28     51     A  Hyderabad
5     Aarav    Male  Web Development   28     96     D     Mumbai
..      ...     ...              ...  ...    ...   ...        ...
145   Anaya    Male  Cloud Computing   19     44     B  Bangalore
146   Meera  Female          AI & ML   18     57     D  Hyderabad
147   Kabir  Female  Cloud Computing   21     92     F  Bangalore
148   Sanya  Female   Cyber Security   26     60     D      Delhi
149    Diya  Female          AI & ML   24     44     F      Delhi

[149 rows x 7 columns]

In [95]:
df.duplicated().sum() # shows the duplicate in dataset

0

**The Dataset doesn't have any duplicate values.**

In [98]:
df.drop_duplicates(inplace=True) # to drop the duplicates values